apps/cowswap-frontend/src/common/hooks/usePendingApprovalModal.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react'
+
+import { getIsMetaMask, getWeb3ReactConnection, injectedConnection } from '@cowprotocol/wallet'
+import { useWeb3React } from '@web3-react/core'
+
+import { useModalState } from './useModalState'
+
+import { MediumAndUp, useMediaQuery } from '../../legacy/hooks/useMediaQuery'
+import { ConfirmationPendingContent } from '../pure/ConfirmationPendingContent'
+import { MetamaskApproveBanner } from '../pure/MetamaskApproveBanner'
+
+function useIsMetaMaskDesktop(): boolean {
+  const { connector } = useWeb3React()
+  const connectionType = getWeb3ReactConnection(connector)
+  const isMetaMask = getIsMetaMask()
+  const isNotMobile = useMediaQuery(MediumAndUp)
+
+  return isMetaMask && isNotMobile && connectionType === injectedConnection
+}
+
+export function usePendingApprovalModal(currencySymbol?: string, onDismiss?: () => void) {
+  const state = useModalState<string>()
+  const { closeModal, context } = state
+
+  const isMetaMaskDesktop = useIsMetaMaskDesktop()
+
+  const onDismissCallback = () => {
+    closeModal()
+    onDismiss?.()
+  }
+
+  const Modal = (
+    <ConfirmationPendingContent
+      onDismiss={onDismissCallback}
+      title={
+        <>
+          Approving <strong>{currencySymbol || context}</strong> for trading
+        </>
+      }
+      description="Approving token"
+      operationLabel="token approval"
+      CustomBody={isMetaMaskDesktop ? <MetamaskApproveBanner /> : undefined}
```

Comment:
I was not too aware of this, we want to show the modal only for Metamask? for other wallets, we show something simpler?

Reply:
<img width="533" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/ce618576-f2a3-4f5a-bacd-39fd19d88369">

Yes, for other wallets we display simplier content

Reply:
ok, makes sense..